src/rnn/types.h

```diff
@@ -83,8 +89,24 @@ class States {
   States select(const std::vector<IndexType>& selIdx, // [beamIndex * activeBatchSize + batchIndex]
                 int beamSize, bool isBatchMajor) const {
     States selected;
+    Expr indices;
+    // I think this doesn't work if model split among gpus but not sure if it matters
+    
+    for (auto& state : states_) {
+      if (state.cell) {
+        indices = state.cell->graph()->indices(selIdx);
+        break;
+      }
+
+      if (state.output) {
+        indices = state.output->graph()->indices(selIdx);
+        break;
+      }
```

Comment:
What happens if neither? Is that a valid condition? If not, let's change this to `else`

Reply:
I think neither is a valid condition

Reply:
Ah, because it's a loop, sorry. But what happens if it never matches any of the conditions? Then `indices` ends up being NULL. Is it worth to `ABORT_IF` for that?

Reply:
Good point. I think `indices` can only end up being NULL if all of the states' `output` and `cell` fields are null. In that case, we will return a vector of nulls which was the behavior of the original code. (Also, I think these values are NULL on the first run of a network so we want this behavior).